app/code/Magento/GraphQl/Model/Query/Logger/NewRelic.php

```diff
@@ -41,10 +41,6 @@ public function __construct(
      */
     public function execute(array $queryDetails)
     {
-        if (!$this->config->isNewRelicEnabled()) {
```

Comment:
@convenient Now you are basically allowing GraphQL logs to be written to New Relic, EVEN if New Relic is disabled in the admin panel. How does that make sense? In a regular scenario, if someone wants to use New Relic, they will enable it in the admin panel as well.

And even if you want to remove code, make sure to remove it properly, i.e. remove the injected dependency as well [here](https://github.com/magento/magento2/blob/381f5d111c8680f3ab22635e05c1a6bc792dbdff/app/code/Magento/GraphQl/Model/Query/Logger/NewRelic.php#L32).

Reply:
Many hosts and systems plug new relic in as a standard PHP app and never configure it within the admin panel.

On those systems you get transaction naming, some from new relic framework detection, some from magento.

Here is an example of where Magento sets the transaction name without checking if the admin panel flag is set
https://github.com/magento/magento2/blob/28c7f79f4d74564ba880e35e59ea45b896437ca7/app/code/Magento/NewRelicReporting/Plugin/CommandPlugin.php#L56-L60

this is because the inner of that function checks if the extension is enabled, which is all that is necessary to set a transaction name